R/generate_dataset.R

```diff
@@ -0,0 +1,68 @@
+#' Generate a dataset
+#' 
+#' This function contains the complete pipeline for generating a dataset
+#' with \pkg{dyngen}. In order to have more control over how the dataset 
+#' is generated, run each of the steps in this function separately.
+#' 
+#' @param model A dyngen initial model created with [initialise_model()].
+#' @param output_dir If not `NULL`, then the generated model and dynwrap 
+#'   dataset will be written to files in this directory.
+#' @param make_plots Whether or not to generate an overview of the dataset.
+#' @param store_grn Whether or not to store the gene regulatory network
+#'   in the dynwrap object as well.
+#' 
+#' @export
+#' @importFrom patchwork wrap_plots plot_annotation
+#' @importFrom ggplot2 ggsave
+generate_dataset <- function(model, output_dir = NULL, make_plots = FALSE, store_grn = FALSE) {
+  model <- model %>% 
+    generate_tf_network() %>% 
+    generate_feature_network() %>% 
+    generate_kinetics() %>% 
+    generate_gold_standard() %>%  
+    generate_cells() %>% 
+    generate_experiment()
+  
+  dataset <-
+    wrap_dataset(model, store_grn = store_grn)
```

Comment:
Might it be worthwile to add some prior information here based on the gold standard?

Reply:
true!